### PR TITLE
Migrate to kube_codegen.sh

### PR DESCRIPTION
**Pull request by matzew**
As per title

### CODE REVIEW

**Comment by matzew** [comment]
I do get these at the end (but is part of update-deps):

```
E0903 09:28:35.452140  516702 library.go:117] Package crypto/subtle does not have module info. Non go modules projects are no longer supported. For feedback, refer to https://github.com/google/go-licenses/issues/128.
F0903 09:28:37.626831  516702 main.go:77] some errors occurred when loading direct and transitive dependency packages
exit status 1
--- FAIL: go-licenses failed the license check
Command '__go_update_deps_for_module' failed in module /home/matzew/go/src/knative.dev/eventing-ceph: 1
```

File: hack/update-codegen.sh [hunk 1]
```diff
@@ -29,14 +30,9 @@ echo "GOPATH=$GOPATH"
...
 
-# generate the code with:
-# --output-base    because this script should also be able to run inside the vendor dir of
-#                  k8s.io/kubernetes. The output-base is needed for the generators to output into the vendor dir
-#                  instead of the $GOPATH directly. For normal projects this can be dropped.
-${CODEGEN_PKG}/generate-groups.sh "deepcopy,client,informer,lister" \
-  "knative.dev/eventing-ceph/pkg/client" "knative.dev/eventing-ceph/pkg/apis" \
-  "sources:v1alpha1" \
-  --go-header-file ${REPO_ROOT_DIR}/hack/boilerplate.go.txt
+kube::codegen::gen_helpers \
```
Comment (creydr) [inline hack/update-codegen.sh:33]: You need to create the clients too (via `kube::codegen::gen_client`). Check for example on https://github.com/knative/eventing/blob/07d379056bd898134b8b5021e66590cea0f16397/hack/update-codegen.sh#L40-L45

**Comment by creydr** [comment]
> I do get these at the end (but is part of update-deps):
> 
> ```
> E0903 09:28:35.452140  516702 library.go:117] Package crypto/subtle does not have module info. Non go modules projects are no longer supported. For feedback, refer to https://github.com/google/go-licenses/issues/128.
> F0903 09:28:37.626831  516702 main.go:77] some errors occurred when loading direct and transitive dependency packages
> exit status 1
> --- FAIL: go-licenses failed the license check
> Command '__go_update_deps_for_module' failed in module /home/matzew/go/src/knative.dev/eventing-ceph: 1
> ```

I got it on your initial version too. I think this was related to the missing clients and some issues before...

**Review by creydr — APPROVED** [review]
Tested via:
```
rm -rf pkg/client
find pkg/apis -name 'zz_generated*' -delete
./hack/update-codegen.sh
```
no diff

/lgtm

**Pull request merged**